spec/lib/services/nodes-api-service-spec.js

```diff
@@ -227,25 +186,268 @@ describe("Http.Services.Api.Nodes", function () {
         });
 
-        it("_removeEnclRelations should remove relations", function() {
-            return nodeApiService._removeEnclRelations(computeNode, [enclosureNode])
+        it("_removeRelation should remove related id", function() {
+            var enclRelationAfter = _.cloneDeep(enclosureNode.relations);
+            _.pull(enclRelationAfter[0].targets, computeNode.id);
+
+            return nodeApiService._removeRelation(enclosureNode, 'encloses', computeNode.id)
+            .then(function () {
+                expect(updateByIdentifier).to.have.been
+                    .calledWith(enclosureNode.id,
+                                {relations: enclRelationAfter});
+            });
+        });
+
+        it("_removeRelation should remove one relation when no target", function() {
+            var enclNode = {
+                id: '1234abcd1234abcd1234abcf',
+                type: 'enclosure',
+                relations: [
+                    {
+                        "relationType": "encloses",
+                        "targets": [
+                            "1234abcd1234abcd1234abcd",
+                        ]
+                    },
+                    {
+                        "relationType": "clusterBy",
+                        "targets": [
+                            "aaa",
+                        ]
+                    }
+                ]
+            };
+            var enclRelationAfter = _.cloneDeep(enclNode.relations);
+            _.pull(enclRelationAfter, enclRelationAfter[0]);
+            return nodeApiService._removeRelation(enclNode, 'encloses', computeNode.id)
             .then(function () {
-                expect(waterline.nodes.updateByIdentifier).to.have.been.calledOnce;
+                expect(updateByIdentifier)
+                   .to.have.been.calledWith(enclNode.id,
+                                            {relations: enclRelationAfter});
             });
         });
 
-        it("_removeEnclRelations should remove enclosure node when it has no relations", function() {
-            return nodeApiService._removeEnclRelations(computeNode, [enclosureNode1])
+        it("_removeRelation should remain empty relation field when no relation", function() {
+            var enclNode = {
+                id: '1234abcd1234abcd1234abcf',
+                type: 'enclosure',
+                relations: [
+                    {
+                        "relationType": "encloses",
+                        "targets": [
+                            "1234abcd1234abcd1234abcd",
+                        ]
+                    }
+                ]
+            };
+            var enclRelationAfter = [];
+            return nodeApiService._removeRelation(enclNode, 'encloses', computeNode.id)
             .then(function () {
+                expect(updateByIdentifier)
+                    .to.have.been.calledWith(enclNode.id,
+                                             {relations: enclRelationAfter});
```

Comment:
This case verifies the relation field in enclNode will be empty after updating, so I check the input parameter for updateByIdentifier().